tests/http/sensors/test_http.py

```diff
@@ -54,29 +53,23 @@ def test_http_run_now_sensor_async(self, mock_poke):
 
         assert isinstance(exc.value.trigger, HttpTrigger), "Trigger is not a HttpTrigger"
 
-    def test_http_response_check_does_not_run_async(self):
-        """Asserts that a task is not deferred when response_check arg is passed to HttpSensorAsync."""
-
+    @mock.patch("airflow.sensors.base.BaseSensorOperator.execute")
+    def test_sensor_not_defer(self, mock_execute):
```

Comment:
Do we not need to check here that defer is not called?

Reply:
yes, since we do not always defer so it makes sense to check that if defer is not expected it should not defer.

Reply:
maybe I can add that too

Reply:
Added, let see what CI think